lib/s3routes/routes.js

```diff
@@ -52,8 +54,12 @@ function checkBucketAndKey(bucketName, objectKey, method, reqQuery,
             blacklistedPrefixes.object);
         if (!result.isValid) {
             log.debug('invalid object key', { objectKey });
-            return errors.InvalidArgument.customizeDescription('Object key ' +
-            `must not start with "${result.invalidPrefix}".`);
+            if (result.invalidPrefix) {
+                return errors.InvalidArgument.customizeDescription('Object ' +
+                `key must not start with "${result.invalidPrefix}".`);
```

Comment:
Can you adjust the message to `Invalid prefix - object key cannot start with "${result.invalidPrefix}".`